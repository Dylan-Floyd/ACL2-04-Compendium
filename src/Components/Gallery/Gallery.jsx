import React from 'react'
import GalleryItem from '../GalleryItem/GalleryItem.jsx'

import './Gallery.css'

export default function Gallery({ galleryItems }) {
  console.log(galleryItems)
  return (
    <div className='gallery'>
      { galleryItems.map(item => <GalleryItem item={item} key={item.id} />) }
    </div>
  )
}
