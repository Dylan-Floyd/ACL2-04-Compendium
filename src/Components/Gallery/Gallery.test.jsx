import { render } from '@testing-library/react'
import Gallery from './Gallery.jsx'

it('Should render multiple pokemon', () => {
  const { container } = render(<Gallery galleryItems={[
    {
      _id: '5ff4fb7cd89993a89cc65481',
      pokemon: 'primeape',
      id: 63,
      species_id: 57,
      height: 10,
      weight: 320,
      base_experience: 159,
      type_1: 'fighting',
      type_2: 'NA',
      attack: 105,
      defense: 60,
      hp: 65,
      special_attack: 60,
      special_defense: 70,
      speed: 95,
      ability_1: 'vital-spirit',
      ability_2: 'anger-point',
      ability_hidden: 'defiant',
      color_1: '#C03028',
      color_2: 'NA',
      color_f: 'NA',
      egg_group_1: 'ground',
      egg_group_2: 'NA',
      url_image: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/057.png',
      generation_id: 1,
      evolves_from_species_id: '56',
      evolution_chain_id: 24,
      shape_id: 6,
      shape: 'upright',
      pokebase: 'primeape',
      pokedex: 'http://www.pokemon.com/us/pokedex/primeape'
    },
    {
      _id: '5ff4fb7cd89993a89cc65485',
      pokemon: 'poliwhirl',
      id: 67,
      species_id: 61,
      height: 10,
      weight: 200,
      base_experience: 135,
      type_1: 'water',
      type_2: 'NA',
      attack: 65,
      defense: 65,
      hp: 65,
      special_attack: 50,
      special_defense: 50,
      speed: 90,
      ability_1: 'water-absorb',
      ability_2: 'damp',
      ability_hidden: 'swift-swim',
      color_1: '#6890F0',
      color_2: 'NA',
      color_f: 'NA',
      egg_group_1: 'water1',
      egg_group_2: 'NA',
      url_image: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/061.png',
      generation_id: 1,
      evolves_from_species_id: '60',
      evolution_chain_id: 26,
      shape_id: 12,
      shape: 'humanoid',
      pokebase: 'poliwhirl',
      pokedex: 'http://www.pokemon.com/us/pokedex/poliwhirl'
    },
    {
      _id: '5ff4fb7cd89993a89cc65488',
      pokemon: 'abra',
      id: 69,
      species_id: 63,
      height: 9,
      weight: 195,
      base_experience: 62,
      type_1: 'psychic',
      type_2: 'NA',
      attack: 20,
      defense: 15,
      hp: 25,
      special_attack: 105,
      special_defense: 55,
      speed: 90,
      ability_1: 'synchronize',
      ability_2: 'inner-focus',
      ability_hidden: 'magic-guard',
      color_1: '#F85888',
      color_2: 'NA',
      color_f: 'NA',
      egg_group_1: 'humanshape',
      egg_group_2: 'NA',
      url_image: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/063.png',
      generation_id: 1,
      evolves_from_species_id: 'NA',
      evolution_chain_id: 27,
      shape_id: 6,
      shape: 'upright',
      pokebase: 'abra',
      pokedex: 'http://www.pokemon.com/us/pokedex/abra'
    }
  ]} />)

  expect(container).toMatchSnapshot()
})