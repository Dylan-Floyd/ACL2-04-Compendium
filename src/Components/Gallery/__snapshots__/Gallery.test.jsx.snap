// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render multiple pokemon 1`] = `
<div>
  <div
    class="gallery"
  >
    <div
      class="cardBorder"
    >
      <div
        class="pokeItem cloudBG fighting"
      >
        <div
          class="topInfo"
        >
          <span
            class="name"
          >
            Primeape 
          </span>
          <span
            class="hp"
          >
            65HP
          </span>
        </div>
        <div
          class="imageBorder"
        >
          <div
            class="imageBG"
          >
            <img
              alt="primeape"
              class="pokeImage"
              src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/057.png"
            />
          </div>
        </div>
        <div
          class="bottomInfo"
        >
          Fighting 
           Pokemon. Length 
          10
          , Weight: 
          320
           lbs.
        </div>
        <ul
          class="infoList"
        >
          <li>
            Type: 
            Fighting 
          </li>
          <li>
            Speed: 
            95
          </li>
          <li>
            Ability 1: 
            Vital Spirit 
          </li>
          <li>
            Base Pokemon: 
            Primeape 
          </li>
        </ul>
      </div>
    </div>
    <div
      class="cardBorder"
    >
      <div
        class="pokeItem cloudBG water"
      >
        <div
          class="topInfo"
        >
          <span
            class="name"
          >
            Poliwhirl 
          </span>
          <span
            class="hp"
          >
            65HP
          </span>
        </div>
        <div
          class="imageBorder"
        >
          <div
            class="imageBG"
          >
            <img
              alt="poliwhirl"
              class="pokeImage"
              src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/061.png"
            />
          </div>
        </div>
        <div
          class="bottomInfo"
        >
          Water 
           Pokemon. Length 
          10
          , Weight: 
          200
           lbs.
        </div>
        <ul
          class="infoList"
        >
          <li>
            Type: 
            Water 
          </li>
          <li>
            Speed: 
            90
          </li>
          <li>
            Ability 1: 
            Water Absorb 
          </li>
          <li>
            Base Pokemon: 
            Poliwhirl 
          </li>
        </ul>
      </div>
    </div>
    <div
      class="cardBorder"
    >
      <div
        class="pokeItem cloudBG psychic"
      >
        <div
          class="topInfo"
        >
          <span
            class="name"
          >
            Abra 
          </span>
          <span
            class="hp"
          >
            25HP
          </span>
        </div>
        <div
          class="imageBorder"
        >
          <div
            class="imageBG"
          >
            <img
              alt="abra"
              class="pokeImage"
              src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/063.png"
            />
          </div>
        </div>
        <div
          class="bottomInfo"
        >
          Psychic 
           Pokemon. Length 
          9
          , Weight: 
          195
           lbs.
        </div>
        <ul
          class="infoList"
        >
          <li>
            Type: 
            Psychic 
          </li>
          <li>
            Speed: 
            90
          </li>
          <li>
            Ability 1: 
            Synchronize 
          </li>
          <li>
            Base Pokemon: 
            Abra 
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
